watchmal/dataset/cnn_mpmt/cnn_mpmt_segmentation_dataset.py

```diff
@@ -0,0 +1,100 @@
+"""
+Class implementing a mPMT dataset for CNNs with segmentation
+"""
+
+# hydra imports
+from hydra.utils import instantiate
+
+# torch imports
+from torch.utils.data import Dataset
+
+# generic imports
+import numpy as np
+import pickle
+
+# WatChMaL imports
+from watchmal.dataset.h5_dataset import H5TrueDataset
+import watchmal.dataset.data_utils as du
+
+class CNNmPMTSegmentationDataset(Dataset):
+    def __init__(self, digi_dataset_config, true_hits_h5file, digi_truth_mapping_file, valid_parents=(-1, 2, 3),
+                 parent_type="max", transform_segmented_labels = True):
+        """
+        Args:
+            digi_dataset_config     ... config for dataset for digitized hits
+            true_hits_h5file        ... path to h5 dataset file for true hits
+            digi_truth_mapping_file ... path to file with a pickled list mapping digitized hit events to true hit events
+            valid_parents           ... valid ID values for hit parents
+
+        """
+        self.digi_dataset = instantiate(digi_dataset_config)
```

Comment:
Minor comment, but is there a particular reason the digihits needs its own config (I don't think the configs are included here). If so this is fine

Reply:
Yeah so I had to play around a bit to get a way for it to work nicely with hydra and the configs for the two datasets. I think the reason I settled on this way was so that the digi_dataset_config could just use the existing config file we already have for e.g. classification.